subprojects/docs/src/docs/userguide/running-builds/gradle_daemon.adoc

```diff
@@ -258,9 +258,53 @@ This message means that either
 --
 In both cases the build cannot benefit from file system watching.
 
-`Caught exception: Couldn't add watch, error = 28`::
-If you receive this error on Linux then you ran out of inotify handles.
-To raise the limit see https://confluence.jetbrains.com/display/IDEADEV/Inotify+Watches+Limit[here].
-
 `java.io.IOException: Too many open files`::
 If you receive this error on macOS, you need to raise your open files limit, see https://superuser.com/a/443168/8117[here].
+
+===  Linux specific notes
+
+File system watching uses http://en.wikipedia.org/wiki/Inotify[Inotify] on Linux.
+Depending on the size of your build, it may be necessary to increase Inotify limits.
+If you are using an IDE, then you probably already did increase the limits.
+
+[[sec:inotify_watches_limit]]
+Inotify watches limit::
+File system watching uses one Inotify watch per watched directory.
+You can see the current limit of Inotify watches per user by running:
++
+[source,bash]
+----
+cat /proc/sys/fs/inotify/max_user_watches
+----
++
+To increase the limit to e.g. 512K watches:
+
+1. Add the following line to either `/etc/sysctl.conf` or a new `*.conf` file under `/etc/sysctl.d/` directory:
```

Comment:
I'd just give one option (to edit `/etc/sysctl.conf`, people who want to create a new file are probably already familiar with how you can do that.

I'd even prefer just this:

```text
echo 524288 | sudo tee /proc/sys/fs/inotify/max_user_watches
sudo sysctl -p --system
```

The main point is that I think it should be possible for most users to copy-paste instructions that would work, and get on with their lives.

Reply:
We could even add a command to increase `max_user_instances`, though I don't think doing that is necessary.

Reply:
I changed the instructions.